schedview/param.py

```diff
@@ -0,0 +1,97 @@
+# Subclasses of param.Parameter for use in schedview.
+
+import param
+import pandas as pd
+
+
+class TypedParameter(param.Parameter):
+    """A parameter with a type.
+
+    Parameters
+    ----------
+    `value_type`: `type`
+        The type the parameter must have.
+    """
+
+    __slots__ = ["value_type"]
+
+    def __init__(self, default=None, value_type=None, allow_None=False, **kwargs):
+        super().__init__(default=default, allow_None=allow_None, **kwargs)
+        self.value_type = value_type
+        self.allow_None = default is None or allow_None
+        self._validate(default)
+
+    def _validate_value(self, val, allow_None):
+        if allow_None and val is None:
+            return
+
+        if not isinstance(val, self.value_type):
+            raise ValueError(
+                f"Parameter {self.name} only takes a {self.value_type}, "
+                f"not value of type {type(val)}."
+            )
+
+    def _validate(self, val):
+        self._validate_value(val, self.allow_None)
```

Comment:
Nice! I was just trying to find out how to do this looking at the documentation, since I was thinking that the juggling between astropy.Time and datetime.time may not be necessary in the prenight app (if you made a Time parameter and used that, which would probably show up looking nicer on screen).